src/scripts/components/widgets/modals/Modal.js

```diff
@@ -65,23 +81,58 @@ const SVG = styled.svg`
 
 const Content = styled.div`
   height: 100%;
-  padding: 40px 46px;
+  padding: 44px 48px;
   width: 100%;
 `
 
+export const ModalContentWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  width: 100%;
+`
+
+export const ModalScrollContent = styled.div`
+  flex: 1 1 100%;
+
+  @media (max-width: 767px) {
+    overflow-x: hidden;
+    overflow-y: scroll;
+  }
+
+  @media (max-height: 767px) {
+    overflow-x: hidden;
+    overflow-y: scroll;
+  }
+`
+
 class Modal extends Component<Props, void> {
+  renderModal () {
+    const { modalContent } = this.props
+    switch (modalContent) {
+      case 'ModalStake':
+        return <ModalStake />
+    }
+  }
   render () {
+    const isVisible = this.props.showModal
+    document.onkeydown = event => {
```

Comment:
This needs to be done in `componentDidMount`. Otherwise, every time this component renders we add this event listener.

Then in `componentWillUnmount` you should remove the event listener

Reply:
And if you just set `.onkeydown` without using `document.on('keydown')` isn't this going to override any other keydown listeners we have set?

Reply:
@bent0b0x Yeah! Its is override 
 I'll change this

Reply:
https://github.com/Paratii-Video/paratii-portal/pull/301/commits/49f681e2a5dc7801ec9e0e4aa422af395946a54b and https://github.com/Paratii-Video/paratii-portal/pull/301/commits/bf833c6419be3b0e190bfc60e4e270d764f41157